op-node/rollup/engine/events.go

```diff
@@ -25,12 +20,7 @@ var ReplaceBlockSource = eth.L1BlockRef{
 	Time:       0,
 }
 
-type Metrics interface {
-	CountSequencedTxsInBlock(txns int, deposits int)
-
-	RecordSequencerBuildingDiffTime(duration time.Duration)
-	RecordSequencerSealingTime(duration time.Duration)
-}
+// no local metrics interface; engine depends directly on op-node/metrics.Metricer
```

Comment:
If the EngineController does not use the whole interface, a smaller interface seems better. Am I missing something?

Reply:
This PR is merging two structs - EngDeriver and EngController into one struct. These two were using two different metrics interfaces, and Teddy merged them into one.

I am not sure what is the benefit of having separate interfaces for metrics for sub-components. Could make sense if we test every single metric we emit, but this seems like an overkill, given the current refactor size.